[4/x][AMD][Lowering Enablement] Enabling meta internal AOTInductor compilation on ROCM

zoranzhao opened the pull request:
Summary: as title

Test Plan: CI & unit test

Differential Revision: D56163334




cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

chenyang78 commented on torch/_inductor/codecache.py:
```diff
@@ -1513,24 +1526,38 @@ def get_include_and_linking_paths(
+        # we need to put each entire -I list despite the redundancies
+        if torch.version.hip is not None:
+            ipaths.append(build_paths.openmp())
+            ipaths.append(build_paths.python())
```
nit - can we put the common ones like ```openmp```, ```python``` to the outside of ```if-else``` branches? Thanks.

zoranzhao replied:
@chenyang78 Updated, please take another look. Since the order of include paths do matter, so a result, we need to have several branches interleaved here.

chenyang78 commented on torch/_inductor/codecache.py:
```diff
@@ -1373,18 +1373,28 @@ def homebrew_libomp() -> Tuple[bool, str]:
+            pass
+    else:
+        # No special treatment needed for OSS
+        pass
```
super nit - perhaps we could simplify the code to the following?

```
def _set_gpu_runtime_env() -> None:
     if (
         config.is_fbcode() and
         torch.version.hip is None and
         "CUDA_HOME" not in os.environ and
         "CUDA_PATH" not in os.environ
    ):
        os.environ["CUDA_HOME"] = os.path.dirname(build_paths.cuda())
```

zoranzhao committed:
[4/x][AMD][Lowering Enablement] Enabling meta internal AOTInductor compilation on ROCM (#124123)

Summary:

as title

Test Plan:
CI & unit test

patch on top of https://www.internalfb.com/phabricator/paste/view/P1214895953 to test

Differential Revision: D56163334

chenyang78 reviewed (approved):
LGTM. Thanks!
